plugins/user-settings/src/components/AuthProviders/ProviderSettingsItem.tsx

```diff
@@ -49,10 +50,20 @@ export const ProviderSettingsItem = (props: {
   const { title, description, icon: Icon, apiRef } = props;
 
   const api = useApi(apiRef);
+  const identityApi = useApi(identityApiRef);
   const errorApi = useApi(errorApiRef);
   const [signedIn, setSignedIn] = useState(false);
   const [profile, setProfile] = useState<ProfileInfo>(emptyProfile);
-
+  const handleLogout = () => {
+    const provider = localStorage.getItem(
+      '@backstage/core:SignInPage:provider',
+    );
```

Comment:
Reading this value here breaks the abstraction of the sign-in page and it won't work for some custom sign-in pages. Do you have any other ideas for how we could identity when an auth API is also used as the identity API?

Reply:
yes, I will think about the abstraction bit.
But can you give me an example of custom sign-in pages or link docs?
because I thought if someone wants to set their custom signing provider then they will add their attributes in identityProviders.
for ex = we using aws as an auth login 
  {
    id: 'aws-auth-provider',
    title: 'aws',
    message: 'Sign In using aws',
    apiRef: awsAuthApiRef,
  }
then save providers. id in local storage.

Reply:
Docs on the sign-in page component are over here: https://backstage.io/docs/auth/#sign-in-configuration

There's no explicit example on how to create a completely custom page, just mentioned in the text.